cmd/thermal-recorder/main.go

```diff
@@ -132,25 +135,78 @@ func runMain() error {
 	}
 }
 
+type HeaderInfo struct {
+	resX      int
+	resY      int
+	fps       int
+	framesize int
+	model     string
+	brand     string
+}
+
+func (h *HeaderInfo) ResX() int {
+	return h.resX
+}
+func (h *HeaderInfo) ResY() int {
+	return h.resY
+}
+func (h *HeaderInfo) FPS() int {
+	return h.fps
+}
+
+func NewHeader(headerInfo map[string]interface{}) *HeaderInfo {
+	return &HeaderInfo{
+		resX:      headerInfo[headers.XResolution].(int),
+		resY:      headerInfo[headers.YResolution].(int),
+		fps:       headerInfo[headers.FPS].(int),
+		framesize: headerInfo[headers.FrameSize].(int),
+		model:     headerInfo[headers.Model].(string),
+		brand:     headerInfo[headers.Model].(string),
+	}
+}
+
+func readHeader(conn net.Conn) (*HeaderInfo, error) {
+	var buf bytes.Buffer
+	scanner := bufio.NewScanner(conn)
+	for {
+		if ok := scanner.Scan(); !ok {
+			break
+		}
+		line := scanner.Text()
+		if strings.Trim(line, " ") == "" {
+			break
+		}
+		buf.WriteString(line + "\n")
+	}
+	header := make(map[string]interface{})
+	err := yaml.Unmarshal(buf.Bytes(), &header)
+	return NewHeader(header), err
+}
+
 func handleConn(conn net.Conn, conf *Config) error {
 
 	totalFrames := 0
-
-	cptvRecorder := NewCPTVFileRecorder(conf)
+	header, err := readHeader(conn)
+	if err != nil {
+		return err
+	}
+	cptvRecorder := NewCPTVFileRecorder(conf, header, header.brand, header.model)
 	defer cptvRecorder.Stop()
 	var recorder recorder.Recorder = cptvRecorder
 
 	if conf.Throttler.Activate {
 		minRecordingLength := conf.Recorder.MinSecs + conf.Recorder.PreviewSecs
-		recorder = throttle.NewThrottledRecorder(cptvRecorder, &conf.Throttler, minRecordingLength, new(throttle.ThrottledEventRecorder))
+		recorder = throttle.NewThrottledRecorder(cptvRecorder, &conf.Throttler, minRecordingLength, new(throttle.ThrottledEventRecorder), header)
 	}
 
-	processor = motion.NewMotionProcessor(&conf.Motion, &conf.Recorder, &conf.Location, nil, recorder)
-
-	rawFrame := new(lepton3.RawFrame)
+	processor = motion.NewMotionProcessor(&conf.Motion, &conf.Recorder, &conf.Location, nil, recorder, header)
 
 	log.Print("new camera connection, reading frames")
 
+	frameLogIntervalFirstMin *= header.FPS()
+	frameLogInterval *= header.FPS()
+
+	rawFrame := new(lepton3.RawFrame)
 	for {
 		_, err := conn.Read(rawFrame[:])
```

Comment:
We'll need to use `io.ReadFull` here to read exact number of bytes. Right now it's still working because the frame fits within a single packet (<64K) but it won't once the larger Boson frames are sent.

https://golang.org/pkg/io/#ReadFull

Reply:
I couldn't not see anything about a limit in the docs, and it appears to work when i test with frames of 691200 size. I would of thought as it's streaming it would just split the packets and recreate them when needed?